reference_data/management/tests/update_gencode_tests.py

```diff
@@ -0,0 +1,159 @@
+            call_command('update_gencode')
+        self.assertEqual(ce.exception.message, 'Error: argument --gencode-release is required')
+
+        # Test required argument out-of-range
+        with self.assertRaises(CommandError) as ce:
+            call_command('update_gencode', '--gencode-release=18')
+        self.assertEqual(ce.exception.message, 'Error: argument --gencode-release: invalid choice: 18 (choose from 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31)')
+
+        # Test genome_version out-of-range
+        with self.assertRaises(CommandError) as ce:
+            call_command('update_gencode', '--gencode-release=19', '/var/tmp', '39')
+        self.assertEqual(ce.exception.message, "Error: argument genome_version: invalid choice: u'39' (choose from '37', '38')")
+
+        # Test missing genome_version when a GTF file is provided
+        with self.assertRaises(CommandError) as ce:
+            call_command('update_gencode', '--gencode-release=19', '/var/tmp')
+        self.assertEqual(ce.exception.message, "The genome version must also be specified after the gencode GTF file path")
+
+        # Test gencode_release and genome_version mis-matched case 1
+        mock_isfile.return_value = True
+        with self.assertRaises(CommandError) as ce:
+            call_command('update_gencode', '--gencode-release=19', '/var/tmp.gz', '38')
+        mock_isfile.assert_called_with('/var/tmp.gz')
+        self.assertEqual(ce.exception.message, "Invalid genome_version: 38. gencode v19 only has a GRCh37 version")
+
+        # Test gencode_release and genome_version mis-matched case 2
+        mock_isfile.reset_mock()
+        mock_isfile.return_value = True
+        with self.assertRaises(CommandError) as ce:
+            call_command('update_gencode', '--gencode-release=20', '/var/tmp1.gz', '37')
+        mock_isfile.assert_called_with('/var/tmp1.gz')
+        self.assertEqual(ce.exception.message, "Invalid genome_version: 37. gencode v20, v21, v22 only have a GRCh38 version")
+
+        # Test genome_version != 38 requires lifted data
+        mock_isfile.reset_mock()
+        mock_isfile.return_value = True
+        with self.assertRaises(CommandError) as ce:
+            call_command('update_gencode', '--gencode-release=23', '/var/tmp2.gz', '37')
+        mock_isfile.assert_called_with('/var/tmp2.gz')
+        self.assertEqual(ce.exception.message, "Invalid genome_version for file: /var/tmp2.gz. gencode v23 and up must have 'lift' in the filename or genome_version arg must be GRCh38")
+
+        # Test --reset option and wrong number data feilds in a line
+        temp_bad_file_path = os.path.join(self.test_dir, 'bad.gencode.v23lift37.annotation.gtf.gz')
+        with gzip.open(temp_bad_file_path, 'w') as f:
+            f.write(u''.join(BAD_FIELDS_GTF_DATA))
+        with self.assertRaises(ValueError) as ve:
+            call_command('update_gencode', '--reset', '--gencode-release=23', temp_bad_file_path, '37')
```

Comment:
why does reset have any impact on this test? The bad file should fail regardless of whether the db gets reset so adding it just makes the test confusing

Reply:
Removed the reset option.